### PR TITLE
[Web] Updated babel preset and webpack config

### PR DESCRIPTION
**Pull request by EvanBacon**
Moved from #3545

### CODE REVIEW

File: apps/native-component-list/webpack/webpack.prod.js [hunk 1]
```diff
@@ -25,38 +25,49 @@ module.exports = merge(common, {
...
-			minRatio: 0.8
-		})
+      asset: '[path].br[query]',
+      test: /\.(js|css)$/,
+      threshold: 1024,
+      minRatio: 0.8,
+    }),
+
+    /** Copy the PWA manifest.json and the caching policy serve.json from the template folder to the build folder  */
+    new CopyWebpackPlugin([
```
Comment (ide) [inline apps/native-component-list/webpack/webpack.prod.js]: Any reason not to do this in the common config since we want a manifest there too?

Comment (EvanBacon) [reply apps/native-component-list/webpack/webpack.prod.js]: These are copying to the `web-build` folder which AFAIK isn't used for dev builds

**Comment by fson** [comment]
Getting an error in this branch when running `yarn run web:dev` or `yarn run web:build`:
```
Could not resolve "expo/build/web/Image/AssetSourceResolver" in file /Users/ville/Projects/expo/apps/native-component-list/node_modules/expo/node_modules/expo-asset/src/Asset.js.
```

File: packages/babel-preset-expo/index.js [hunk 2]
```diff
@@ -1,31 +1,150 @@
...
+      [
+        'babel-plugin-module-resolver',
+        {
+          alias: {
+            'react-native-vector-icons': '@expo/vector-icons',
+            /** Alias direct react-native imports to react-native-web */
+            'react-native$': 'react-native-web',
+            /** Add polyfills for modules that react-native-web doesn't support */
+            'react-native/Libraries/Image/AssetSourceResolver$':
+              'expo/build/web/Image/AssetSourceResolver',
```
Comment (fson) [inline packages/babel-preset-expo/index.js]: Looks like this file has moved to `expo-asset/build/AssetSourceResolver`.

File: packages/babel-preset-expo/index.js [hunk 3]
```diff
@@ -1,31 +1,150 @@
...
+            'react-native/Libraries/Image/assetPathUtils$': 'expo/build/web/Image/assetPathUtils',
```
Comment (fson) [inline packages/babel-preset-expo/index.js]: Looks like this file has been removed in #3545.

File: packages/babel-preset-expo/index.js [hunk 4]
```diff
@@ -1,31 +1,150 @@
...
+            'react-native/Libraries/Image/resolveAssetSource$':
+              'expo/build/web/Image/resolveAssetSource',
```
Comment (fson) [inline packages/babel-preset-expo/index.js]: This file seems to have moved to `expo-asset/build/resolveAssetSource`.

**Comment by EvanBacon** [comment]
Sorry I forgot to rebase after merging the image polyfill

File: packages/babel-preset-expo/index.js [hunk 5]
```diff
@@ -1,31 +1,150 @@
...
+            'react-native/Libraries/Image/AssetSourceResolver$':
+              'expo-asset/build/AssetSourceResolver',
+            'react-native/Libraries/Image/assetPathUtils$': 'expo-asset/build/Image/assetPathUtils',
+            'react-native/Libraries/Image/resolveAssetSource$':
+              'expo-asset/build/resolveAssetSource',
+          },
+        },
+      ],
+      ['@babel/plugin-syntax-flow'],
+      ['@babel/plugin-proposal-optional-catch-binding'],
```
Comment (ide) [inline packages/babel-preset-expo/index.js]: We shouldn't need most of these plugins since babel-preset-env has them (I sent you the list of them)

**Review by ide — CHANGES_REQUESTED** [review]

**Review by ide — APPROVED** [review]

**Pull request merged**